plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -168,3 +170,52 @@ func WindowsEventLogLevelName(levelId int32) string {
 		return UNKNOWN
 	}
 }
+
+// In some cases wevtapi does not insert values when formatting the message. The message
+// will contain insertion string placeholders, of the form %n, where %1 indicates the first
+// insertion string, and so on. Noted that wevtapi start the index with 1.
+// https://learn.microsoft.com/en-us/windows/win32/eventlog/event-identifiers#insertion-strings
+//
+// If we see those data in the `EventData` section, `insertPlaceholderValues` format the message
+// with the correct values
+func insertPlaceholderValues(rawMessage string, evtDataValues []Data) string {
+	if len(evtDataValues) == 0 {
```

Comment:
nit: Consider checking the `rawMessage` for `%` as well to see if we can skip this entire section.

Reply:
That's true. I was thinking checking `%` would iterate through the strings first and iterate again when inserting values. This would make it a O(2n) time. But I guess although it would be O(n) without checking `%`, re-constructing the string for `rawMessage` without `%` also takes time/resouce

Reply:
2nd thought on this. I'm keeping the current implementation for rawMessage without `%` since :
- it will jump only do
   ```
   sb.WriteString(rawMessage[0:])
   ```
   after iterating through `rawMessage` only once and return `sb.String()`
- So in the case there are not any `%`, it's doing almost the same as `!strings.Contains(rawMessage, "%")`. But in case there are any `%`, it is going to iterate each character of `rawMessage` twice.